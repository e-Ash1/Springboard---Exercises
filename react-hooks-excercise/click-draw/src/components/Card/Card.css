.deck-container {
    position: relative;
    width: 200px; 
    height: 300px;
    overflow: hidden;
  }
  
  .card {
    position: absolute;
    top: 0;
    left: 0;
    transition: all 0.3s ease;
  }
  