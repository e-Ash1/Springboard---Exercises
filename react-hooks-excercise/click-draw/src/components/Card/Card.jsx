import React from 'react';
import './Card.css';

const Card = ({ image }) => {
    if (!image) return null;
  
    return (
    <div>
        <img src={image} alt="Playing Card" />
    </div>
    );

  };

export default Card;