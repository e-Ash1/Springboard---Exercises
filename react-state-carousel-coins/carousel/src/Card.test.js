import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';


//Smoke Test:
it('renders without crashing', ()=>{
    render(<Card />)
});

//Snapshot Test:
it('matches snapshot', ()=>{
    const { asFragment } = render(
        <Card
            caption='test caption'
            src='test.jpg'
            currNum={1}
            totalNum={3}
        />
    );
    expect(asFragment()).toMatchSnapshot();

});

