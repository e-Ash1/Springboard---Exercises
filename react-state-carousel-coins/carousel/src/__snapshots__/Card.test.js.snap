// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test caption
    </h4>
    <img
      alt="test caption"
      class="Card-image"
      src="test.jpg"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;
